src/main/java/de/codingair/tradesystem/tradelog/TradeLogService.java

```diff
@@ -0,0 +1,36 @@
+package de.codingair.tradesystem.tradelog;
+
+import de.codingair.tradesystem.TradeSystem;
+import de.codingair.tradesystem.tradelog.repository.TradeLogRepository;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+
+import java.util.List;
+
+public class TradeLogService {
+
+    private static TradeLogService instance;
+
+    private TradeLogRepository tradeLogRepository = TradeSystem.getInstance().getTradeLogRepository();
+
+    public static TradeLogService getTradeLog() {
+        if (instance == null) {
+            instance = new TradeLogService();
+        }
+        return instance;
+    }
+
+    private TradeLogService() {
+    }
+
+    public void log(Player player1, Player player2, String message) {
+        Bukkit.getScheduler().runTask(TradeSystem.getInstance(), () -> {
```

Comment:
I feel very bad to request another change, but this is still a synchronized task. Please use runTaskAsynchronously()

Reply:
Aw damn sorry man, I'm still fairly new to this Minecraft plugin development. I'll change it. I assumed running a task would be asynchronous

Reply:
Ok changed it